File: samples/button/reference/swift/Tokens.swift
import UIKit

private extension UIColor {
    convenience init(hex: UInt32) {
        self.init(
            red: CGFloat((hex >> 16) & 0xff) / 255,
            green: CGFloat((hex >> 8) & 0xff) / 255,
            blue: CGFloat(hex & 0xff) / 255,
            alpha: CGFloat((hex >> 24) & 0xff) / 255
        )
    }
}

enum ColorToken {
    static let brand500: UIColor = UIColor(hex: 0xff1a73e8)
    static let buttonPrimaryBg: UIColor = UIColor { trait in
        trait.userInterfaceStyle == .dark ? UIColor(hex: 0xff8ab4f8) : ColorToken.brand500
    }
    static let buttonPrimaryFg: UIColor = UIColor { trait in
        trait.userInterfaceStyle == .dark ? ColorToken.neutral900 : ColorToken.neutral0
    }
    static let buttonSecondaryBg: UIColor = UIColor { trait in
        trait.userInterfaceStyle == .dark ? UIColor(hex: 0xff303134) : ColorToken.neutral100
    }
    static let buttonSecondaryFg: UIColor = UIColor { trait in
        trait.userInterfaceStyle == .dark ? ColorToken.neutral0 : ColorToken.neutral900
    }
    static let neutral0: UIColor = UIColor(hex: 0xffffffff)
    static let neutral100: UIColor = UIColor(hex: 0xfff1f3f4)
    static let neutral900: UIColor = UIColor(hex: 0xff1f1f1f)
}

enum RadiusToken {
    static let md: CGFloat = 8
}

enum SpaceToken {
    static let md: CGFloat = 16
    static let sm: CGFloat = 8
}

enum TypoToken {
    static let labelMdFontFamily: String = "Inter"
    static let labelMdFontSize: CGFloat = 14
    static let labelMdFontWeight: CGFloat = 600
    static let labelMdLineHeight: CGFloat = 20
}

File: samples/button/reference/kotlin/Tokens.kt
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

object ColorToken {
    val brand500: Color = Color(0xff1a73e8)
    val buttonPrimaryBg: Color
        @Composable get() = if (isSystemInDarkTheme()) Color(0xff8ab4f8) else ColorToken.brand500
    val buttonPrimaryFg: Color
        @Composable get() = if (isSystemInDarkTheme()) ColorToken.neutral900 else ColorToken.neutral0
    val buttonSecondaryBg: Color
        @Composable get() = if (isSystemInDarkTheme()) Color(0xff303134) else ColorToken.neutral100
    val buttonSecondaryFg: Color
        @Composable get() = if (isSystemInDarkTheme()) ColorToken.neutral0 else ColorToken.neutral900
    val neutral0: Color = Color(0xffffffff)
    val neutral100: Color = Color(0xfff1f3f4)
    val neutral900: Color = Color(0xff1f1f1f)
}

object RadiusToken {
    val md: Dp = 8.dp
}

object SpaceToken {
    val md: Dp = 16.dp
    val sm: Dp = 8.dp
}

object TypoToken {
    val labelMdFontFamily: String = "Inter"
    val labelMdFontSize: TextUnit = 14.sp
    val labelMdFontWeight: Float = 600f
    val labelMdLineHeight: TextUnit = 20.sp
}
